### DISCUSSION TITLE
Week 1: NASA EarthData Access

### DISCUSSION THREAD

**Discussion by eeholmes** [UW-Upwelling-Project/Summer-2023#2] [Announcements]
Week 1:

* Learn how to access NASA Earth Data in the cloud by working through the 2021 Cloud Hackathon tutorials

https://nasa-openscapes.github.io/2021-Cloud-Hackathon/tutorials/00_Setup.html

## Steps

1. Log into our JupyterHub. https://jhub.opensci.live/
2. Fork the Cloud Hackathon repo to your own GitHub account https://github.com/nasa-openscapes/2021-Cloud-Hackathon
3. Here is a project to try to run https://github.com/oceanhackweek/ohw21-proj-cmip-ard

**Comment by eeholmes** [comment]
Briefly describe research:

For the summer internship description: "we using machine learning along with remote-sensing data to study climate change in the Indian Ocean and Bay of Bengal".  During my summer internship, I am working with my fellow intern to develop a data base for this project and then for OceanHackWeek, I will pitch an idea on this work.

**Comment by minhphan03** [comment]
Hi @eeholmes do you have the new jupyter hub setup that you said would be correctly configured yet? I am going on a very long flight and am probably going to be very tired so I don't think I can start on Monday just yet. Is it OK if I can start on Tuesday instead? Let me know if you have anything new going on I need to know before we start.

**Reply by eeholmes** [reply]
@minhphan03 Still struggling with getting the hub set-up so no worries re starting Tuesday! We'll have a check in on Tuesday at 5pm Seattle time with Jiaurui.

**Comment by NaNa7Miiii** [comment]
Hi @eeholmes, may I ask are we going to do check-in via zoom today?

**Reply by eeholmes** [reply]
Yes at 5pm. I will send a zoom link

**Comment by eeholmes** [comment]
Zoom link: https://washington.zoom.us/j/95552498139

**Comment by eeholmes** [comment]
Issues with hub

* Still can't get dask working
* missing pip install zarr-eosdis-store
```
/tmp/ipykernel_481/303224034.py:10: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display
  from IPython.core.display import display, HTML
```

**Comment by NaNa7Miiii** [comment]
Issue with hub (access deny):
I tried to use the code below to access the data:
**s3_cred_endpoint = {
    'podaac':'https://archive.podaac.earthdata.nasa.gov/s3credentials',
    'lpdaac':'https://data.lpdaac.earthdatacloud.nasa.gov/s3credentials'
}
def get_temp_creds():
    temp_creds_url = s3_cred_endpoint['podaac']
    return requests.get(temp_creds_url).json()
temp_creds_req = get_temp_creds()**
, and got **JSONDecodeError: [Errno Expecting value] HTTP Basic: Access denied.**
I have double-checked that my .netrc file is indeed stores in my jhub.

**Comment by eeholmes** [comment]
Yes I am trying to sort out the problem. Hopefully sorted by 5pm check in

**Comment by eeholmes** [comment]
More tutorials on NASA Earthdata

https://nasa-openscapes.github.io/earthdata-cloud-cookbook/
This shows how to access the data wo S3 but with https using https://pypi.org/project/earthaccess/
https://nasa-openscapes.github.io/earthdata-cloud-cookbook/external/Sea_level_rise_notebook.html
More examples using earthdata
https://github.com/betolink/earthaccess-gallery/tree/main/notebooks

**Comment by eeholmes** [comment]
@minhphan03 @NaNa7Miiii Ok, I might have a better hub up and running. Has 32 Gig RAM and 100Gi of storage. Later I will figure out have to create a shared drive for the data files so we all use the same directory.

https://dhub.opensci.live/

You'll need to set up GitHub:

# Steps

1. Copy your personal access token from GitHub (scopes repo and user)
2. Log into the JupyterHub and open a terminal
```
git config --global user.name "Your name"
git config --global user.email "your@email.com"
git config --global credential.helper store
```
You will need to use your GitHub username and the personal access token the first time that you push a change to GitHub.

# Create your `.netrc` file again for using Earthdata

Post any issues in the thread here.

**Reply by NaNa7Miiii** [reply]
Thank you! Got the set up!

**Reply by eeholmes** [reply]
Great! I need to figure out how to connect a shared drive for the data we/you are assembling but I'll need to get an Azure support person to help. I couldn't figure out all the pieces.

**Comment by eeholmes** [comment]
I have a repo for the Docker image for workspace

https://github.com/nmfs-opensci/nmfs-jhub/tree/main/ci/iopython

Put in an [issue](https://github.com/nmfs-opensci/nmfs-jhub/issues) or pull request if there are new packages to add and I will update.
